src/main/java/org/hobbit/core/components/channel/ReadByteChannel.java

```diff
@@ -0,0 +1,86 @@
+package org.hobbit.core.components.channel;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import com.rabbitmq.client.AMQP.BasicProperties;
+/**
+ * For each readByte call for DirectChannel implementation,
+ * a thread of ReadByteChannel is created. This makes sure that
+ * the execution takes place on a separate thread by waiting for 
+ * messages to be received by the pipe.
+ * @author altaf, sourabh, yamini, melisa
+ *
+ */
+public class ReadByteChannel extends DirectChannel implements Runnable{
```

Comment:
It seems like this class is never used, or am I missing something?

Reply:
ReadByteChannel is used in readBytes() method of DirectChannel class.